diff --git i/rails/guides/rails_guides/markdown.rb w/guides/rails_guides/markdown.rb
index d926d233bb..ced64acd34 100644
--- i/rails/guides/rails_guides/markdown.rb
+++ w/rails/guides/rails_guides/markdown.rb
@@ -109,7 +109,7 @@ def generate_structure
                   hierarchy = hierarchy[0, 3] + [node]
                 end
 
-                node[:id] = dom_id(hierarchy) unless node[:id]
+                node[:id] = dom_id(hierarchy).parameterize unless node[:id]
                 node.inner_html = "#{node_index(hierarchy)} #{node.inner_html}"
               end
             end
diff --git i/rails/guides/rails_guides/markdown/renderer.rb w/guides/rails_guides/markdown/renderer.rb
index 8e560da515..34ba573254 100644
--- i/rails/guides/rails_guides/markdown/renderer.rb
+++ w/rails/guides/rails_guides/markdown/renderer.rb
@@ -19,7 +19,7 @@ def block_code(code, language)
           <div class="code_container">
           <pre><code class="highlight #{lexer_language(language)}">#{formatted_code}</code></pre>
           <textarea class="clipboard-content" id="#{clipboard_id}">#{clipboard_content(code, language)}</textarea>
-          <button class="clipboard-button" data-clipboard-target="##{clipboard_id}">Copy</button>
+          <button class="clipboard-button" data-clipboard-target="##{clipboard_id}">Copiar</button>
           </div>
         HTML
       end
